components/tikv_kv/src/lib.rs

```diff
@@ -111,19 +111,48 @@ pub struct WriteData {
     pub modifies: Vec<Modify>,
     pub extra: TxnExtra,
     pub deadline: Option<Deadline>,
+    pub disk_full_opt: DiskFullOpt,
 }
 
 impl WriteData {
     pub fn new(modifies: Vec<Modify>, extra: TxnExtra) -> Self {
+        Self::new_ext(modifies, extra, None, DiskFullOpt::NotAllowedOnFull)
```

Comment:
These changes about constructors of `WriteData` are not necessary. Just keep the old code, and add a new method `set_disk_full_opt` is fine enough.